webapp/channels/src/components/suggestion/search_channel_with_permissions_provider.tsx

```diff
@@ -159,10 +152,10 @@ export default class SearchChannelWithPermissionsProvider extends Provider {
 
         const channelsAsync = this.autocompleteChannelsForSearch(teamId, channelPrefix);
 
-        let channelsFromServer = [];
+        let channelsFromServer: Channel[] = [];
         try {
             const {data} = await channelsAsync;
-            channelsFromServer = data;
+            channelsFromServer = data!;
```

Comment:
How do you know data is always defined here?

Reply:
I just assumed it was always defined because it worked before. Now I wonder if code like this is actually how `undefined` entries sometimes get added to arrays because `['channel1', 'channel2'].concat(undefined)` results in `['channel1', 'channel2', undefined]` for some awful reason.

Regardless, I'll fix the type here and ensure that this value is actually always an array